Tests/Contour.RabbitMq.Tests/MetricCollectingSpecs.cs

```diff
@@ -0,0 +1,156 @@
+﻿using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+using Common.Logging;
+
+using Contour.Receiving;
+using Contour.Testing.Transport.RabbitMq;
+using Contour.Transport.RabbitMQ;
+using Contour.Transport.RabbitMQ.Topology;
+
+using FluentAssertions;
+
+using NUnit.Framework;
+
+namespace Contour.RabbitMq.Tests
+{
+    /// <summary>
+    /// The basic request response specs.
+    /// </summary>
+    [SuppressMessage("StyleCop.CSharp.NamingRules", "SA1300:ElementMustBeginWithUpperCaseLetter", Justification = "Reviewed. Suppression is OK here.")]
+    public class MetricCollectingSpecs
+    {
+        /// <summary>
+        /// The when_requesting_using_custom_callback_route.
+        /// </summary>
+        [TestFixture]
+        [Category("Integration")]
+        public class when_handling_request_message : RabbitMqFixture
+        {
+            #region Public Methods and Operators
+
+            /// <summary>
+            /// The should_return_response.
+            /// </summary>
+            [Test]
+            public void should_collect_metrics_with_correct_tags()
```

Comment:
This integration test must be replaced by a set of unit tests for the each client of the metric collector.

Reply:
@mzabolotko, one of the clients is Listener which does not have unit tests because of it's complex implementation. Should I write Listener's unit test just for testing metric collector work or shall we leave integration test?